### DISCUSSION TITLE
[RFC] Pass arbitrary metadata to turborepo remote cache events

### DISCUSSION THREAD

**Discussion by finn-orsini** [vercel/turborepo#7268] [Ideas]
### Describe the feature you'd like to request

We would like a way to pass arbitrary metadata to the remote cache endpoints, so we can collect more specific metrics around the cache hits / misses. 

We are currently able to use the `events` endpoint to log the cache hit / miss, hash specified, and `local` v. `remote` cache hit from the `source` value - but we would like to distinguish further. 

For example, we would like a way to signify where the request came from: i.e. a local request, a development pipeline, a production pipeline, or an ephemeral env pipeline. For the pipeline cases, we would also like to record information about the specific pipeline / action (i.e. a pipeline ID and specific build/job ID). 

### Describe the solution you'd like

One option may be to support arbitrary environment variables with a particular prefix, i.e. anything named `TURBOREPO_REMOTE_CACHE_<KEY>` added to the payload. 

Ideally the solution would not be opinionated about the specific keys / data added to the events payload, and be easy to change between build envs.

### Describe alternatives you've considered

Open to discussion / suggestion.

**Comment by anthonyshew** [comment]
Some more thoughts on this same idea on a duplicate issue here: https://github.com/vercel/turbo/issues/2802
